lms/djangoapps/courseware/model_data.py

```diff
@@ -3,8 +3,8 @@
 """
 
 import json
+from abc import abstractmethod, ABCMeta
```

Comment:
Could you please add a module level comment giving a very brief overview of the different classes in this module, their relationship, and when you would give an alternate implementation (e.g. why is there a `DjangoOrmFieldCache` instead of a generic cache over specific kv-stores?).